llama-server: fix duplicate HTTP headers in multiple models mode

ServeurpersoCom opened the pull request:
*Make sure to read the [contributing guidelines](https://github.com/ggml-org/llama.cpp/blob/master/CONTRIBUTING.md) before submitting a PR*

## Approach: Filter at source

This patch filters headers before forwarding them to avoid duplication.

Why headers get duplicated:
When the router proxies child process responses, both the router (via 
set_default_headers) and the child send the same headers (Server, 
Transfer-Encoding, Keep-Alive, CORS). The proxy was forwarding everything, 
resulting in duplicates.

Solution:
1. Skip headers that will be added by the router or httplib:
   - server, transfer-encoding, keep-alive (duplicated by router defaults)
   - access-control-* (CORS headers injected by router)

2. Handle Content-Type separately via msg_t.content_type to avoid duplication 
   when httplib calls set_chunked_content_provider() or set_content().

Tested with:

```
curl -v http://127.0.0.1:8082/v1/chat/completions   -H "Content-Type: application/json"   -d '{
    "model": "Codestral-22B-v0.1-i1-GGUF",
    "messages": [{"role": "user", "content": "hi"}],
    "max_tokens": 5
  }'
*   Trying 127.0.0.1:8082...
* Connected to 127.0.0.1 (127.0.0.1) port 8082 (#0)
> POST /v1/chat/completions HTTP/1.1
> Host: 127.0.0.1:8082
> User-Agent: curl/7.88.1
> Accept: */*
> Content-Type: application/json
> Content-Length: 121
>
< HTTP/1.1 200 OK
< Keep-Alive: timeout=5, max=100
< Transfer-Encoding: chunked
< Access-Control-Allow-Origin:
< Server: llama.cpp
< Content-Length: 591
< Content-Type: application/json; charset=utf-8
< Connection: close
<
* Closing connection 0
{"choices":[{"finish_reason":"length","index":0,"message":{"role":"assistant","content":" Hello! I'm"}}],"created":1764699011,"model":"Codestral-22B-v0.1-i1-GGUF","system_fingerprint":"b7274-f57a165eb","object":"chat.completion","usage":{"completion_tokens":5,"prompt_tokens":5,"total_tokens":10},"id":"chatcmpl-6WGgrUU4EWxLfKAXfzUq9hVDLrbybfJm","timings":{"cache_n":4,"prompt_n":1,"prompt_ms":21.409,"prompt_per_token_ms":21.409,"prompt_per_second":46.7093278527722,"predicted_n":5,"predicted_ms":51.717,"predicted_per_token_ms":10.343399999999999,"predicted_per_second":96.68000850784075}}
```

Before: duplicate Server, Transfer-Encoding, Keep-Alive, Access-Control-Allow-Origin, Content-Type
After: all headers appear exactly once

Fixes #17693

ServeurpersoCom committed:
llama-server: fix duplicate HTTP headers in multiple models mode (#17693)

ngxson commented on tools/server/server-models.cpp:
```diff
@@ -968,7 +1005,10 @@ server_http_proxy::server_http_proxy(
-        this->headers = header.headers;
+        this->headers = std::move(header.headers);
+        if (!header.content_type.empty()) {
+            this->content_type = std::move(header.content_type);
```
If we std::move here, I prefer restrict the scope of `header` to avoid someone (in the future) accidentally use `header` after this point

ngxson reviewed (commented):
looks good overall! would appropriate if you can address some small comments

ServeurpersoCom committed:
llama-server: address review feedback from ngxson

- restrict scope of header after std::move
- simplify header check (remove unordered_set)

ServeurpersoCom commented on tools/server/server-models.cpp:
```diff
@@ -889,6 +892,32 @@ struct pipe_t {
+        "server",
+        "transfer-encoding",
+        "keep-alive"
+    };
```
unordered_set replaced with if()

ngxson reviewed (approved):
nice, thanks! (merging once the CI passes)
